lib/compat/wordpress-6.2/rest-api.php

```diff
@@ -108,3 +108,43 @@ function gutenberg_modify_rest_sidebars_response( $response ) {
 	return $response;
 }
 add_filter( 'rest_prepare_sidebar', 'gutenberg_modify_rest_sidebars_response' );
+
+
+/**
+ * Add the `block_types` value to the `pattern-directory-item` schema.
+ *
+ * @since 6.2.0 Added 'block_types' property.
+ */
+function add_block_pattern_block_types_schema() {
+	register_rest_field(
+		'pattern-directory-item',
+		'block_types',
+		array(
+			'schema' => array(
+				'description' => __( 'The block types which can use this pattern.', 'gutenberg' ),
+				'type'        => 'array',
+				'uniqueItems' => true,
+				'items'       => array( 'type' => 'string' ),
+				'context'     => array( 'view', 'embed' ),
+			),
+		)
+	);
+}
+add_filter( 'rest_api_init', 'add_block_pattern_block_types_schema' );
```

Comment:
We already extend `WP_REST_Pattern_Directory_Controller` in 6.2 compat dir. Maybe we should change the schema and response there.

What do you think?

Reply:
When we can update something with filters, I think it's preferable as we don't need to copy the whole function.

Reply:
I don't have any strong preference here. Having all contextual changes in a single place makes backports easier.